tests/unit/CoreHeadingTest.php

```diff
@@ -0,0 +1,627 @@
+				'lock'            => null,
+				'placeholder'     => null,
+				'style'           => null,
+				'textAlign'       => null,
+				'textColor'       => 'contrast-2',
+			],
+			$attributes
+		);
+	}
+
+	public function test_retrieve_core_heading_with_gradient() {
+		$block_content = '
+		<!-- wp:heading {"gradient":"gradient-3","fontSize":"medium"} -->
+		<h2 class="wp-block-heading has-gradient-3-gradient-background has-background has-medium-font-size">hello</h2>
+		<!-- /wp:heading -->
+		';
+
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$actual = graphql(
+			[
+				'query'     => $this->query(),
+				'variables' => [ 'id' => $this->post_id ],
+			]
+		);
+
+		$block      = $actual['data']['post']['editorBlocks'][0];
+		$attributes = $block['attributes'];
+
+		$this->assertEquals(
+			[
+				'align'           => null,
+				'anchor'          => null,
+				'backgroundColor' => null,
+				'className'       => null,
+				'content'         => 'hello',
+				'cssClassName'    => 'wp-block-heading has-gradient-3-gradient-background has-background has-medium-font-size',
+				'fontFamily'      => null,
+				'fontSize'        => 'medium',
+				'gradient'        => 'gradient-3', // Previously untested
+				'level'           => 2.0,
+				'lock'            => null,
+				'placeholder'     => null,
+				'style'           => null,
+				'textAlign'       => null,
+				'textColor'       => null,
+			],
+			$attributes
+		);
+	}
+
+	public function test_retrieve_core_heading_with_custom_gradient() {
+		$block_content = '
+			<!-- wp:heading {"style":{"color":{"gradient":"linear-gradient(135deg,rgb(6,147,227) 0%,rgb(155,81,224) 100%)"}}} -->
+			<h2 class="wp-block-heading has-background" style="background:linear-gradient(135deg,rgb(6,147,227) 0%,rgb(155,81,224) 100%)">Gradient Heading</h2>
+			<!-- /wp:heading -->
+		';
+
+		wp_update_post(
+			[
+				'ID'           => $this->post_id,
+				'post_content' => $block_content,
+			]
+		);
+
+		$actual = graphql(
+			[
+				'query'     => $this->query(),
+				'variables' => [ 'id' => $this->post_id ],
+			]
+		);
+
+		$block      = $actual['data']['post']['editorBlocks'][0];
+		$attributes = $block['attributes'];
+
+		$this->assertEquals(
+			[
+				'align'           => null,
+				'anchor'          => null,
+				'backgroundColor' => null,
+				'className'       => null,
+				'content'         => 'Gradient Heading',
+				'cssClassName'    => 'wp-block-heading has-background',
+				'fontFamily'      => null,
+				'fontSize'        => null,
+				'gradient'        => null,
+				'level'           => 2.0,
+				'lock'            => null,
+				'placeholder'     => null,
+				'style'           => wp_json_encode(
+					[
+						'color' => [
+							'gradient' => 'linear-gradient(135deg,rgb(6,147,227) 0%,rgb(155,81,224) 100%)',
+						],
+					]
+				),
+				'textAlign'       => null,
+				'textColor'       => null,
+			],
+			$attributes
+		);
+	}
```

Comment:
We can remove this, we're already testing `style`, the internal props don't matter.